android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -440,6 +440,13 @@ class GeckoEngineSession(
         }
     }
 
+    /**
+     * See [EngineSession.hasCookieBannerRuleForSession]
+     */
+    override fun hasCookieBannerRuleForSession(): Boolean {
+        return geckoSession.hasCookieBannerRuleForBrowsingContextTree().poll()!!
```

Comment:
As `hasCookieBannerRuleForBrowsingContextTree` is async call we will need to expose  a way for consumers to get the value without blocking, we could use callbacks you can find an similar example in [GeckoEngine](https://github.com/mozilla-mobile/firefox-android/blob/68f6309018768326683fa16a45ccc0c046ac5226/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L226-L263).